src/components/filterSelection/FilterSelection.js

```diff
@@ -13,7 +18,70 @@ export const Status = {
 };
 
 const FilterSelection = ({ label, status, amount, onClick, onClearClick }, ref) => {
-  return <Box ref={ref}>Hello FilterSelection!</Box>;
+  const handleClearClick = useCallback(
```

Comment:
Again, don't memoize be default, you should only memoize inline functions if they turn out to be really expensive. It can be cheaper just to create functions every time than to memoize them